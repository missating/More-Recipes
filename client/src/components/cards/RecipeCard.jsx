import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


const RecipeCard = props => (
  <div className="flex">

    <div className="card">
      <Link to={`/recipes/view/${props.id}`} className="hover">
        <img
          src={props.recipeImage}
          alt={props.recipeImage}
          className="image-width"
        />
      </Link>
      <div className="container p-3 recipedetails-column">
        <h2>{props.name}</h2>
        <p><small>{props.description}</small></p>
        <div className="center-button">
          <Link
            className="btn btn-secondary"
            to={`/recipes/view/${props.id}`}
          >
            VIEW
          </Link>
          {
            props.button &&
            (
              <button
                id="removeButton"
                className="btn btn-secondary"
                onClick={() => props.onButtonClick(props.id)}
              >
                Remove
              </button>
            )

          }
        </div>
      </div>
    </div>
  </div>
);


RecipeCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  recipeImage: PropTypes.string.isRequired,
  onButtonClick: PropTypes.func,
  button: PropTypes.bool
};

RecipeCard.defaultProps = {
  onButtonClick: () => null,
  button: false
};

export default RecipeCard;
