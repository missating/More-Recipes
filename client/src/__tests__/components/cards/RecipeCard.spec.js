import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import RecipeCard from '../../../components/cards/RecipeCard';

describe('Recipe card component', () => {
  const props = {
    id: 1,
    name: 'test',
    description: 'add all',
    recipeImage: 'image_url',
  };


  it('Should render', () => {
    const wrapper = shallow(<RecipeCard {...props} />);
    expect(wrapper.find('div').length).toEqual(4);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Should remove favourite when button is clicked', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<RecipeCard
      {...props}
      onButtonClick={spy}
      button
    />);
    wrapper.find('#removeButton').simulate('click');
    expect(spy.called).toBeTruthy();
  });
});

