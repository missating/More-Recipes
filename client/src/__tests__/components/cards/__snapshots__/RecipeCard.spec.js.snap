// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe card component Should render 1`] = `
<div
  className="flex"
>
  <div
    className="card"
  >
    <Link
      className="hover"
      replace={false}
      to="/recipes/view/1"
    >
      <img
        alt="image_url"
        className="image-width"
        src="image_url"
      />
    </Link>
    <div
      className="container p-3 recipedetails-column"
    >
      <h2>
        test
      </h2>
      <p>
        <small>
          add all
        </small>
      </p>
      <div
        className="center-button"
      >
        <Link
          className="btn btn-secondary"
          replace={false}
          to="/recipes/view/1"
        >
          VIEW
        </Link>
      </div>
    </div>
  </div>
</div>
`;
